worlds/spelunker/__init__.py

```diff
@@ -0,0 +1,145 @@
+        self.locked_locations = []
+
+    def create_item(self, name: str) -> Item:
+        data = item_table[name]
+        return Item(name, data.classification, data.code, self.player)
+
+    def create_regions(self) -> None:
+        init_areas(self, get_locations(self))
+
+    def get_filler_item_name(self) -> str:
+        if self.random.randint(0, 100) <= 10:
+            return self.random.choice(useful_items)
+        else:
+            return self.random.choice(filler_items)
+
+    def set_rules(self) -> None:
+        set_rules(self)
+        self.multiworld.completion_condition[self.player] = lambda state: state.has("Golden Pyramid", self.player)
+        self.get_location("Golden Pyramid").place_locked_item(self.create_item("Golden Pyramid"))
+
+    def get_excluded_items(self) -> Set[str]:
+        excluded_items: Set[str] = set()
+        return excluded_items
+
+    def create_item_with_correct_settings(self, name: str) -> Item:
+        data = item_table[name]
+        item = Item(name, data.classification, data.code, self.player)
+        return item
+
+    def generate_filler(self, pool: List[Item]) -> None:
+        for _ in range(len(self.multiworld.get_unfilled_locations(self.player)) - len(pool) - 1):
+            item = self.create_item_with_correct_settings(self.get_filler_item_name())
+            pool.append(item)
+
+    def get_item_pool(self, excluded_items: Set[str]) -> List[Item]:
+        pool: List[Item] = []
+
+        for name, data in item_table.items():
+            if name not in excluded_items:
+                for _ in range(data.amount):
+                    item = self.create_item_with_correct_settings(name)
+                    pool.append(item)
+
+        return pool
+
+    def create_items(self) -> None:
+        pool = self.get_item_pool(self.get_excluded_items())
+
+        self.generate_filler(pool)
+
+        self.multiworld.itempool += pool
+
+    def generate_output(self, output_directory: str):
+        try:
+            world = self.multiworld
+            player = self.player
+            patch = SpelunkerProcPatch()
+            patch_rom(self, patch, self.player, self.multiworld)
+
+            self.rom_name = patch.name
+
+            patch.write(os.path.join(output_directory,
+                                     f"{self.multiworld.get_out_file_name_base(self.player)}{patch.patch_file_ending}"))
+        except Exception:
+            raise
+        finally:
+            self.rom_name_available_event.set()  # make sure threading continues and errors are collected
+
+    def modify_multidata(self, multidata: dict) -> None:
+        # wait for self.rom_name to be available.
+        self.rom_name_available_event.wait()
+        rom_name = getattr(self, "rom_name", None)
+        if rom_name:
+            new_name = base64.b64encode(bytes(self.rom_name)).decode()
+            multidata["connect_names"][new_name] = multidata["connect_names"][self.multiworld.player_name[self.player]]

```

Comment:
You can use the new helper function
```suggestion
            multidata["connect_names"][new_name] = multidata["connect_names"][self.player_name]

```